src/game.cc

```diff
@@ -1193,71 +1201,139 @@ static void gameFreeGlobalVars()
 // 0x443F74
 static void showHelp()
 {
-    ScopedGameMode gm(GameMode::kHelp);
+    ScopedGameMode gm(GameMode::kHelp); // Switch to Help game mode.
 
     bool isoWasEnabled = isoDisable();
     gameMouseObjectsHide();
-
     gameMouseSetCursor(MOUSE_CURSOR_NONE);
 
     bool colorCycleWasEnabled = colorCycleEnabled();
     colorCycleDisable();
 
-    // CE: Help screen uses separate color palette which is incompatible with
-    // colors in other windows. Setup overlay to hide everything.
-    int overlay = windowCreate(0, 0, screenGetWidth(), screenGetHeight(), 0, WINDOW_HIDDEN | WINDOW_MOVE_ON_TOP);
+    // load the optional stretch setting from f2_res.ini
+    int helpScreenStretchMode = 0;
+    int stretchGameMode = 0;
+    Config config;
+    if (configInit(&config)) {
+        if (configRead(&config, "f2_res.ini", false)) {
+            configGetInt(&config, "STATIC_SCREENS", "HELP_SCRN_SIZE", &helpScreenStretchMode);
+
+            if (configGetInt(&config, "STATIC_SCREENS", "STRETCH_GAME", &stretchGameMode)) {
+                helpScreenStretchMode = stretchGameMode; // always override if key exists
+            }
+        }
+        configFree(&config);
+    }
+
+    int screenWidth = screenGetWidth();
+    int screenHeight = screenGetHeight();
+
+    // Create an invisible fullscreen overlay window to cover everything.
+    int overlay = windowCreate(0, 0, screenWidth, screenHeight, 0, WINDOW_HIDDEN | WINDOW_MOVE_ON_TOP);
+
+    FrmImage helpFrmImage;
+    int helpFid = buildFid(OBJ_TYPE_INTERFACE, 297, 0, 0, 0); // Load Help screen graphic (FID 297).
+    if (!helpFrmImage.lock(helpFid)) {
+        if (colorCycleWasEnabled)
+            colorCycleEnable();
+        gameMouseObjectsShow();
+        if (isoWasEnabled)
+            isoEnable();
+        return;
+    }
+
+    unsigned char* helpData = helpFrmImage.getData();
+    int helpWidth = HELP_SCREEN_WIDTH;
+    int helpHeight = HELP_SCREEN_HEIGHT;
+
+    unsigned char* bufferToBlit = helpData;
+    int blitWidth = helpWidth;
+    int blitHeight = helpHeight;
+
+    // Check if stretching is needed (based on config or screen too small).
+    if (helpScreenStretchMode != 0 || screenWidth < helpWidth || screenHeight < helpHeight) {
+        int scaledWidth, scaledHeight;
+
+        calculateScaledSize(
+            helpWidth,
+            helpHeight,
+            screenWidth,
+            screenHeight,
+            helpScreenStretchMode,
+            scaledWidth,
+            scaledHeight);
 
-    int helpWindowX = (screenGetWidth() - HELP_SCREEN_WIDTH) / 2;
-    int helpWindowY = (screenGetHeight() - HELP_SCREEN_HEIGHT) / 2;
-    int win = windowCreate(helpWindowX, helpWindowY, HELP_SCREEN_WIDTH, HELP_SCREEN_HEIGHT, 0, WINDOW_HIDDEN | WINDOW_MOVE_ON_TOP);
+        // Allocate memory for the scaled image.
+        unsigned char* scaled = reinterpret_cast<unsigned char*>(internal_malloc((scaledWidth + 1) * (scaledHeight + 1)));
```

Comment:
Why does this need the +1's?  blitBufferToBufferStretchAndFixEdges only copies the exact destination size to my reading.